test/keboola/facebook/api/request_test.clj

```diff
@@ -113,17 +113,23 @@
 (deftest test-get-next-page-url
   (let [response {:paging {:previous "https://graph.facebook.com/v16.0/17841401480255572/insights?access_token=XXX&pretty=0&since=1672915646&until=1675248446&metric=total_interactions&metric_type=total_value&period=day"
                            :next "https://graph.facebook.com/v16.0/17841401480255572/insights?access_token=XXX&pretty=0&since=1677581248&until=1679914048&metric=total_interactions&metric_type=total_value&period=day"}}]
-    (is (nil? (sut/get-next-page-url response true)))
-    (is (= (sut/get-next-page-url response false) "https://graph.facebook.com/v16.0/17841401480255572/insights?access_token=XXX&pretty=0&since=1677581248&until=1679914048&metric=total_interactions&metric_type=total_value&period=day")))
+    (is (nil? (sut/get-next-page-url response true false)))
+    (is (= (sut/get-next-page-url response false false) "https://graph.facebook.com/v16.0/17841401480255572/insights?access_token=XXX&pretty=0&since=1677581248&until=1679914048&metric=total_interactions&metric_type=total_value&period=day")))
 
   (let [response {:paging {:previous "https://graph.facebook.com/v16.0/17841401480255572/insights?access_token=XXX&pretty=0&since=1672915646&until=1675248446&metric=total_interactions&metric_type=total_value&period=day"}}]
-    (is (nil? (sut/get-next-page-url response true)))
-    (is (nil? (sut/get-next-page-url response false))))
+    (is (nil? (sut/get-next-page-url response true false)))
+    (is (nil? (sut/get-next-page-url response false false))))
+
+  (let [response {:paging {:next "someurl"} :data []}]
+    (is (nil? (sut/get-next-page-url response false true))))
+
+  (let [response {:paging {:next "someurl"} :data [{:some "data"}]}]
+    (is (= "someurl" (sut/get-next-page-url response false true))))
```

Comment:
test case 2 - if stop-on-empty-response is enabled and `data` in the reponse is not empty (mocked as `:data [{:some "data"}]}`) then return `someurl`, and pagination will continue.